src/main/java/uk/gov/companieshouse/insolvency/data/service/InsolvencyService.java

```diff
@@ -1,31 +1,36 @@
 package uk.gov.companieshouse.insolvency.data.service;
 
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 import uk.gov.companieshouse.api.insolvency.CompanyInsolvency;
 import uk.gov.companieshouse.api.insolvency.InternalCompanyInsolvency;
 import uk.gov.companieshouse.api.insolvency.InternalData;
 import uk.gov.companieshouse.insolvency.data.model.InsolvencyDocument;
 import uk.gov.companieshouse.insolvency.data.model.Updated;
 import uk.gov.companieshouse.insolvency.data.repository.InsolvencyRepository;
-import uk.gov.companieshouse.logging.Logger;
 
 @Service
 public class InsolvencyService {
 
-    private final Logger logger;
     private final InsolvencyRepository insolvencyRepository;
 
-    public InsolvencyService(Logger logger, InsolvencyRepository insolvencyRepository) {
-        this.logger = logger;
+    /**
+     * Insolvency service to store insolvency data onto mongodb and call chs kafka endpoint.
+     * @param insolvencyRepository repository to save data to db
+     */
+    public InsolvencyService(InsolvencyRepository insolvencyRepository) {
         this.insolvencyRepository = insolvencyRepository;
+
     }
 
     /**
      * Persist company insolvency information to mongodb collection.
      * @param insolvencyApi company insolvency information {@link InternalCompanyInsolvency}
      */
+    @Transactional
```

Comment:
Transaction should be at controller level to wrap invocation to Chs Kafka API in same transaction, if fails it will rollback the update and error/retry ticket will replay the message